directus-solidstart/index.md

```diff
@@ -0,0 +1,314 @@
+
+## Fetch Global Data
+
+Now that the table is created and we have data added, let's create our first fetcher. In `/lib/directus.js` we will add a new method:
+
+```js
+export const getGlobals = cache(async () => {
+  "use server";
+  try {
+    return await directus.request.readItems("global");
+  } catch (e) {
+    console.error(e);
+
+    return null;
+  }
+}, `globals`);
+```
+
+Notice the `"use server"` directive. This indicates the bundler this method will never run (and therefore will not exist) in the client-side build. The `cache()` method wrapper defines a cache-key ("globals") where this response will be stored. We will try the request to Directus, if the request is not successful we can throw a `404`.
+
+With our fetcher, we can head over to our `src/routes/index.jsx` and create our first loader request.
+
+```jsx
+export const route = {
+  load: () => getGlobals(),
+};
+
+export default function Home() {
+  const pageInfo = createAsync(getGlobals);
+
+  createEffect(() => {
+    if (pageInfo() === null) {
+      throw new Error("404");
+    }
+  });
+
+  return (
+    <main>
+      <Show when={pageInfo()}>{(info) => <h1>{info().title}</h1>}</Show>
+
+      <Show when={pageInfo()}>
+        {(info) => <strong>{info().description}</strong>}
+      </Show>
+    </main>
+  );
+}
+```
+
+The `route` object defines a `load` method that will be called **before** our route is accessed. Then `createAsync` will receive the deferred data as a signal and allow our UI to update. The `<Show>` component is important to make sure we only render our elements once the data has finally arrived at the browser.
+
+## Dynamic Routes With Directus
+
+Create a new collection called pages - make the Primary ID Field a "Manually Entered String" called slug, which will correlate with the URL for the page. For example about will later correlate to the page [localhost:3000/about](http://localhost:3000/about).
+
+Create a text input field called title and a WYSIWYG input field called content. In Roles & Permissions, give the Public role read access to the new collection. Create 3 items in the new collection - here's some sample data.
+
+Now, we go to our `src/lib/directus.js` and create our new fetcher.
+
+```js
+export const getPage = (slug) =>
+  cache(async () => {
+    "use server";
+    try {
+      return await directus.request(readItem("pages", slug));
+    } catch (e) {
+      return null;
+    }
+  }, slug);
+```
+
+Again, this method will attempt to fetch information from a given `slug`. And if there is no information, it will return `null`.
+
+Inside of `src/routes`, create a new a file called `slug.jsx`. This is a dynamic route, so a single file can be used for all of the top-level pages.
+
+> It's important to note that exact matches take precedence over dynamic routes. If it exists, a `about.jsx` file will render instead of `[slug].jsx` for the `/about` URL.
```

Comment:
```suggestion
:::tip Dynamic Route Precedence

It's important to note that exact matches take precedence over dynamic routes. If it exists, a `about.jsx` file will render instead of `[slug].jsx` for the `/about` URL.

:::
```

Reply:
I'd also move this below the code snippet as more of a footnote to the section.